### ISSUE TITLE
Button: `iconSize` prop doesn't work with some icons

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#43745]
### Description

I have found that the size is not applied to some of the `@wordpress/icons` when I specify the `iconSize` attribute on a button component with SVG icons.
These icons will be displayed at the expected size when the size is specified in the `Icons` component, not the `Button` component.

I checked the icon library and found that icons with `width` / `height` attributes of the svg element match those with no size applied. I don't know why these icons don't reproduce when used in the Icons component, but I think SVG icons should be rendered at the size specified in the `Icons` component and the `width` / `height` attributes are unnecessary on the svg element.

The following are those icons:

- [atSymbol](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/at-symbol.js)
- [commentEditLink](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/comment-edit-link.js)
- [commentReplyLink](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/comment-reply-link.js)
- [dragHandle](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/drag-handle.js)
- [helpFilled](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/help-filled.js)

### Step-by-step reproduction instructions

I have rewritten `edit.js` in the code block to include code that designates all icons in `@wordpress/icons` as buttons:

```javascript
import { useBlockProps } from '@wordpress/block-editor';
import * as Icons from '@wordpress/icons';
import { Button, Flex } from '@wordpress/components';

const { Icon, ...availableIcons } = Icons;

export default function CodeEdit() {
	const blockProps = useBlockProps();
	return (
		<div { ...blockProps }>
			<Flex
				wrap="wrap"
				justify="flex-start"
				style={ { textAlign: 'center', fontSize: '8px' } }
			>
				{ Object.entries( availableIcons ).map( ( [ name, icon ] ) => {
					return (
						<div key={ name } style={ { width: '5%' } }>
							<Button icon={ icon } iconSize={ 40 } />
						</div>
					);
				} ) }
			</Flex>
		</div>
	);
}
```

### Screenshots, screen recording, code snippet

The icon circled in red has `width` and `height` specified for the svg element and the `iconSize` property is not applied.

![icons](https://user-images.githubusercontent.com/54422211/187684821-955859fd-f18c-4791-a467-d47b2a2caf10.png)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
I think removing hardcoded sizes should fix the issue. There's #41772, but it's hard to see actual changes because of the bad rebase.

**Comment by t-hamano** [comment]
Oh, I was just about to create #43747 😅
Should we wait for #41772 to be properly rebased?

**Comment by Mamaduka** [comment]
Do feel free to create a new PR 👍  I think that one was only fixing `line-(dashed|dottted|solid).js` files. Maybe fix them as well, since PR hasn't been active for almost 40 days.

**Comment by t-hamano** [comment]
I checked the latest trunk and it appears that those three icons were fixed in another PR, #40315.
Therefore, regarding #41772, it looks like we will have to close it, unfortunately.

Since the problem I have found is different from the icons improved above, I would like to activate my PR.
